### PR TITLE
feat(core-flow): order edit endpoints

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
* The relevant part of this PR is only at folder `packages/medusa/src/api/admin/order-edits/`
All the other changes are Types

Missing: 
* `/store` endpoints to confirm/decline the order change
* http tests of the full flow

### CODE REVIEW

File: packages/medusa/src/api/admin/order-edits/validators.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    id: z.union([z.string(), z.array(z.string())]).optional(),
+    order_id: z.union([z.string(), z.array(z.string())]).optional(),
+    status: z.union([z.string(), z.array(z.string())]).optional(),
+    created_at: createOperatorMap().optional(),
+    updated_at: createOperatorMap().optional(),
+    deleted_at: createOperatorMap().optional(),
+  })
+)
+
+export type AdminGetOrdersParamsType = z.infer<typeof AdminGetOrdersParams>
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/order-edits/validators.ts]:
```suggestion
export type AdminGetOrderEditsParamsType = z.infer<typeof AdminGetOrdersParams>
```

File: packages/medusa/src/api/admin/order-edits/validators.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+export type AdminGetOrdersOrderParamsType = z.infer<
+  typeof AdminGetOrdersOrderParams
+>
+
+/**
+ * Parameters used to filter and configure the pagination of the retrieved order.
+ */
+export const AdminGetOrdersParams = createFindParams({
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/order-edits/validators.ts]:
```suggestion
export const AdminGetOrderEditsParams = createFindParams({
```

File: packages/medusa/src/api/admin/order-edits/validators.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  z.object({
+    id: z.union([z.string(), z.array(z.string())]).optional(),
+    status: z.union([z.string(), z.array(z.string())]).optional(),
+    created_at: createOperatorMap().optional(),
+    updated_at: createOperatorMap().optional(),
+    deleted_at: createOperatorMap().optional(),
+  })
+)
+
+export type AdminGetOrdersOrderParamsType = z.infer<
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/order-edits/validators.ts]:
```suggestion
export type AdminGetOrderEditsParamsType = z.infer<
```

File: packages/medusa/src/api/admin/order-edits/validators.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import { z } from "zod"
+import {
+  createFindParams,
+  createOperatorMap,
+  createSelectParams,
+} from "../../utils/validators"
+
+export const AdminGetOrdersOrderParams = createSelectParams().merge(
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/order-edits/validators.ts]:
```suggestion
export const AdminGetOrderEditsParams = createSelectParams().merge(
```

File: packages/core/core-flows/src/order/workflows/order-edit/begin-order-edit.ts [hunk 5]
```diff
@@ -13,7 +13,7 @@ import { throwIfOrderIsCancelled } from "../../utils/order-validation"
 /**
  * This step validates that an order-edit can be requested for an order.
  */
-export const beginOrderOrderEditValidationStep = createStep(
+export const beginorderEditValidationStep = createStep(
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/order-edit/begin-order-edit.ts:16]:
```suggestion
export const beginOrderEditValidationStep = createStep(
```

File: packages/medusa/src/api/admin/order-edits/query-config.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  "display_id",
+  "order_version",
+  "created_at",
+  "updated_at",
+  "canceled_at",
+]
+
+export const defaultAdminDetailsOrderEditFields = [
+  ...defaultAdminOrderEditFields,
+  "additional_items.*",
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/order-edits/query-config.ts]: question: I presume additional items hold the items added to the order. How are the removed and updated items stored?

Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/admin/order-edits/query-config.ts]: No, this is actually leftover from exchanges. Order edits won't have their own entities.
From the Order change actions we can display all the changes that happened in that specific order change compared to the order version before the change.

Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/admin/order-edits/query-config.ts]: The order edit endpoints should only return the order preview,
returning the order again together with it doesn't bring any value and only increase the response size.

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**